src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -784,9 +784,10 @@ private Variable parseVariable(final Type t, final BasicScope scope) {
     Variable result;
 
     if (Parser.isReservedWord(variableName.content)) {
-      throw this.parseException("Reserved word '" + variableName + "' cannot be a variable name");
+      throw this.parseException(
+          variableName, "Reserved word '" + variableName + "' cannot be a variable name");
     } else if (scope != null && scope.findVariable(variableName.content) != null) {
-      throw this.parseException("Variable " + variableName + " is already defined");
+      throw this.parseException(variableName, "Variable " + variableName + " is already defined");
```

Comment:
Can we not add a proper location here and above? Or do they get added by parseException

Reply:
parseException has multiple overloads that build the Location if it was submitted a Range or a Position, yes

Reply:
(and Token extends Range)